### DISCUSSION TITLE
Registry key not found for registry search of dotnet core runtime

### DISCUSSION THREAD

**Discussion by radz1** [wixtoolset/issues#6293] [Questions] [answered]
I am not able to get the registry key for the dotnet core runtime. 
Registry key not found. Key = 'SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\sharedfx\Microsoft.NETCore.App\3.1.4'

i am using below code in wxs file:

 <util:RegistrySearch Root="HKLM"
                         Key="SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\sharedfx\Microsoft.NETCore.App\3.1.4"
                         Result="value"
                         Variable="NetCore3_x64_Installed"
                         Win64="no" 
                         Format="raw"                                       
                         />

**Comment by robmen** [comment, marked as answer]
You're doing a 32-bit search (`Win64="no"`) so `WOW6432Node` will not be part of the path. Remove `WOW6432Node` or make your search 64-bit.

**Reply by radz1** [reply]
when i am using the key : SOFTWARE\WOW6432Node\dotnet\Setup\InstalledVersions\x64\sharedfx\Microsoft.NETCore.App. It recognize as a key but when adding \3.1.4 in the key, it says key not found. pls refer below image for 32-bit registry folder tree.

Actually, my software has prerequisite of having core runtime 3.1.4 and greater till [3.1.10]. I wanted to get the list of core runtime version and validate before proceed with installation.  For now I achieved it using below code . I have created individual search for each version and put condition if all returns "1" then proceed installation or abort.

<util:RegistrySearch Root="HKLM" Key="[NetCoreKey]" Result="value" Variable="NetCore3_x64_Installed_314" Win64="no" Format="raw" Value="3.1.4"/>
<util:RegistrySearch Root="HKLM" Key="[NetCoreKey]" Result="value" Variable="NetCore3_x64_Installed_315" Win64="no" Format="raw" Value="3.1.5"/>
    <util:RegistrySearch Root="HKLM" Key="[NetCoreKey]" Result="value" Variable="NetCore3_x64_Installed_3110" Win64="no" Format="raw" Value="3.1.10"/>

**Pls suggest if any better way to achieve it.**

![image](https://user-images.githubusercontent.com/14122410/101793534-e9f96400-3b2b-11eb-987d-4bb4f8a88f2e.png)

**Reply by rseanhall** [reply]
There is not really a better way. We hope to get this functionality built in to v4 - https://github.com/wixtoolset/issues/issues/6257 and https://github.com/wixtoolset/issues/issues/6264.

**Reply by radz1** [reply]
Hi @rseanhall , does it fix in new version v4. Do we have the better way to check the .net core versions ?

**Reply by BMurri** [reply]
Looking at that image, `3.1.4` is not a `Key` or a `Value`, It's a `Name`. A search for a key by that path should always fail as you wrote it, based on the contents of your registry.

**Comment by mjanulaitis** [comment]
@rseanhall after two years has this functionality been added?
